lib/ex_unit/lib/ex_unit/server.ex

```diff
@@ -68,6 +77,22 @@ defmodule ExUnit.Server do
       %{config | captured_devices: Set.delete(config.captured_devices, device)}}
   end
 
+  def handle_call({:log_capture_on, pid}, _from, config) do
+    ref = Process.monitor(pid)
+    case config.log_captures do
```

Comment:
We should add the capture backend here as a monitored handler (`GenEvent.add_mon_handler`) so that if it exits the console backend is re-added on receiving the `:gen_event_EXIT`. If the handler crashes (e.g. due to a bad formatter) we can return the error to capture_log where it can raise a suitable error. Also if we monitor/link to the Logger GenEvent we can ensure we always know the state of the handlers on the Logger GenEvent.